R/add_compactness_constraints.R

```diff
@@ -0,0 +1,53 @@
+#' @include internal>R
+NULL
+
+#' Add constraints to limit compactness
```

Comment:
The compactness constraint is a single constraint, I would have named it `add_compactness_constraint`, but I am not a native English speaker, so I might also be missing something here.

Reply:
Ah yeah, that's a good point. Personally, I think the important thing here is consistency - so that all functions for adding a constraint(s) (e.g. `add_compactness_constraint`, `add_components_constraint`, `add_locked_out_constraint`, etc) have the same suffix to indicate that they belong to the same family of functions. I worry that having some of functions ending in `constraints` and others in `constraint` might potentially be confusing for users?  So for example, we could have a `_constraint` suffix, `_constraints` suffix, or something else? How does that sound?

Reply:
I agree, and I would go with the `_constraint` suffix.

Reply:
Ok - sounds good - let's go with that.